### PR TITLE
Start conductr with features in docker scenario

### PR DESCRIPTION
**Pull request by markusjura**
Adds the ability to start the ConductR sandbox cluster with additional features:
- `logging` => Loads and scales `conductr-elasticsearch` to 1 instance
- `visualization` => Loads and scales `visualizer` to 1 instance

In other word the commands `sandbox run` and `sandbox debug` have been now enriched with the option `--with-features`, e.g.:

```
sandbox run --with-features logging
```

OR

```
sandbox debug --with-features visualization logging
```

### CODE REVIEW

**Comment by markusjura** [comment]
Successfully tested manually in combination with `conductr` PR https://github.com/typesafehub/conductr/pull/785.

Ready for review.

File: src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala [hunk 1]
```diff
@@ -150,7 +155,8 @@ object ConductRSandbox extends AutoPlugin {
     envsValue: Map[String, String],
     imageValue: String,
     logLevelValue: String,
-    portsValue: Set[Int]): String = {
+    portsValue: Set[Int],
+    withFeatures: Option[Set[String]]): String = {
```
Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala]: Option should not be needed. An empty set can be regarded as no features right?

Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala]: Ignore - suits the parser.

**Comment by huntc** [comment]
LGTM

**Comment by markusjura** [comment]
Dependent PR https://github.com/typesafehub/conductr/pull/785#issuecomment-134522952 has still and issue. There this PR shouldn't be merged yet.

**Comment by markusjura** [comment]
PR has been fixed in the meantime. Ready to get merged.

**Comment by huntc** [comment]
Jenkins

**Comment by huntc** [comment]
jenkins

**Comment by huntc** [comment]
Doh - no Jenkins here :-)

**Comment by markusjura** [comment]
Jenkins is still failing because of the authentication problems. This should be fixed as soon as the docker image is public.

IMHO the PR is ready to get merged. Tests are passing on my end.

File: src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala [hunk 2]
```diff
@@ -164,13 +174,21 @@ object ConductRSandbox extends AutoPlugin {
 
     val logLevelEnv = Map("AKKA_LOGLEVEL" -> logLevelValue)
     val syslogEnv = cond0Ip.map(ip => Map("SYSLOG_IP" -> ip)).getOrElse(Map.empty)
-    val envsArgs = (logLevelEnv ++ syslogEnv ++ envsValue).map { case (k, v) => s"-e $k=$v" }.toSeq
-
+    val conductrFeaturesArgs =
+      if (featureNames.isEmpty)
```
Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala:178]: Why not map featureNames here?

File: src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala [hunk 3]
```diff
@@ -164,13 +174,21 @@ object ConductRSandbox extends AutoPlugin {
...
+
+    // Expose always the feature related ports even if the are not specified with `--withFeatures`.
+    // Therefor these ports are also exposed if only the `runConductRs` tasks is executed (e.g. in testing)
     val conductrPorts = Set(
       9004,
       9005,
       9006,
-      9200
+      9200, // conductr-elasticsearch
+      9999 // visualizer
```
Comment (huntc) [inline src/main/scala/com/typesafe/conductr/sandbox/sbt/ConductRSandbox.scala]: Perhaps comment the others for consistency.

**Comment by huntc** [comment]
Lgtm Just some small comments

**Comment by huntc** [comment]
LGTM - please merge

**Pull request merged**